src/packet/BinaryWriter.js

```diff
@@ -1,137 +1,62 @@
-﻿'use strict';
+'use strict';
 /*
  * Simple BinaryWriter is a minimal tool to write binary stream with unpredictable size.
  * Useful for binary serialization.
  *
- * Copyright (c) 2016 Barbosik
- * 
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- * 
- *     http://www.apache.org/licenses/LICENSE-2.0
- * 
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- * 
- * 
+ * Copyright (c) 2016 Barbosik https://github.com/Barbosik
+ * License: Apache License, Version 2.0
  */
 
-function BinaryWriter(size) {
-    if (!size || size <= 0) {
-        size = Buffer.poolSize / 2;
-    }
-    this._buffer = new Buffer(size);
+function BinaryWriter() {
+    this._buffer = parseInt(process.version[1]) < 6 ? new Buffer(1e5) : Buffer.allocUnsafe(1e5);

```

Comment:
@Andrews54757

> First off, my tactic is not solvable with two lines of code.
- If that's not your tactic then, what is it? Being a troll?
- I like being minimal in code amount.

> Also since the maximum of cells is around 50000, according to the buffer size, 
- The maximum of cells possible to be at the same time is the limit of an uint32, 4 294 967 296. A size 100 000 buffer can store a minimum of 3 999 cell updates, protocol 5. [Barbosik used a limit of 131 072 (128*1024) for node updating in the old BinaryWriter](https://github.com/Megabyte918/MultiOgar-Edited/blob/master/src/packet/UpdateNodes.js#L15).

> it also sets it the maximum amount of minions and other things.
- **We. Don't. Care.** If you really want to have 4000 minions in a 1920x1080 map then have fun with the errors and CPU time.

> Multi-Ogar's main purpose is to be a server for the masses.
- MultiOgar's main purpose is to be a reliable server, without memory leaks that Ogar has, and faster than Ogar.

> 100 players > 3000 kilobytes of data
- V8 is not stupid enough to keep 98KB of memory unallocated and cause a memory leak.
- 3000KB is still 2.93MB.
- 300 players is still 286MB of buffers. Assuming all players have 300 nodes on their screen 93% of the memory is unused. A single 2GB memory chip will handle the server well.

> And Multi-Ogar's cpu graphs are horrible.
- Still faster than Ogar. And OgarUL.
- It doesn't spread CPU load through the available 40 milliseconds. This makes async handling really easy.

> You can make BinaryWriter two times more efficient by fixing the toBinary code. (Basically, precalculate)
- > Is it fast enough?
  > Yes!
  > Alright, zero changes!
- I can also make MultiOgar 6x faster by handling bot updates and player packet building in a separate child process. But I don't want to.